src/library/scala/collection/Iterable.scala

```diff
@@ -169,6 +169,9 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     */
   def iterableFactory: IterableFactory[IterableCC]
 
+  @deprecated("Use iterableFactory instead", "2.13.0")
+  @`inline` def companion: IterableFactory[IterableCC] = iterableFactory
```

Comment:
Didn't make it `final`, in case if some library ovveridden this method.

Reply:
Well, in such a case the override will have no effects. We should at least notify the user with a warning (by using `@deprecatedOverriding`), or an error (by making it final … but in such a case we can not suggest the user what to override instead).

Reply:
added `@deprecatedOverriding`. Thanks